lib/ansible/modules/files/synchronize.py

```diff
@@ -414,6 +416,8 @@ def main():
         rsync = module.get_bin_path(rsync, required=True)
 
     cmd = [rsync, '--delay-updates', '-F']
+    if rsync_password:
+        cmd = ['sshpass', '-p', rsync_password] + cmd
```

Comment:
Shoot.... There's a problem, this is insecure.  command line args can be seen on nearly any platform so they cannot be used to pass around passwords and other secrets.

Reply:
We will need to follow the pattern as used in the ssh connection plugin of creating a socket and passing the password over the socket.

Reply:
These days environment variables (-e with SSHPASS) are safe on most platforms but there may be some platforms still in use where any user can view those as well.  -f might be okay but we'd have to be very careful not to create a race condition where someone can place their own temporary file or directory to subvert that.  -d, opening up a separate file descriptor, is likely the most secure and least error prone way to pass the password securely but you may have to use subprocess instead of module.run_command to make use of that.  (The call to run_command() looks very simple so I don't believe porting to raw subprocess will be a problem).

Reply:
@abadger good catch, thanks

Reply:
thanks for reviewing, i've pushed the fix

Reply:
@abadger autocheck not allowing to use Popen